src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -888,9 +906,11 @@ private void UpdateCallsiteExecutionState(Object callsiteData, RuntimeCore runti
                         {
                             resolvesFeps = new List<FunctionEndPoint>() { compliantTarget };
                             replicationInstructions = replicationOption;
-                            return;
+                            matchFound = true;
```

Comment:
@reddyashish so not returning here seems to fix this case:

this branch on the left |  1.3.4 on the right
<img width="1610" alt="Screen Shot 2019-03-21 at 9 33 57 PM" src="https://user-images.githubusercontent.com/508936/54795466-bcb3a680-4c22-11e9-95da-52c7b8d5bfe4.png">

I guess because we walk further down through all cases of the possible replications .... does it make sense to reverse the list of replicationsTrials if we will always keep iterating on the list of `replicationInstruction(s)`?

also - I find it odd the performance benchmarks are showing the same speed.... let's try to manufacture a case which we know will be slower - for example:

<img width="804" alt="Screen Shot 2019-03-21 at 9 50 09 PM" src="https://user-images.githubusercontent.com/508936/54795615-51b69f80-4c23-11e9-9176-d51d8a27e85e.png">

produces 25 `replicationInstruction`s to try - before your change this would have exited much earlier - it would be produce all nulls of course... but I want to see if we can get a sense of the difference in speed or bound it.

Reply:
Yes, case4: would represent the cases where there are different ranks of array's or null elements(as this is zero and others are rank 1). From my understanding the difference between case 2 and case 4 is that in case2 it finds an exact match and in case 4 it looks for other matches(like circle object being sent to curve input). In the example you attached, it is working if you fix in case 4 because int is being sent to double. If the input list has double values, it would find the match in case 2 and we will have to check all the replication options there. 
Case 6 takes care of the cases where we have other data types like int, boolean, string(but not in any nested list)  
Example: [3, [c], [[c]], "test"].

Reply:
I had thought of reversing the list but we do not want to do that all the time. The replication options list is sent into this function is in some order that is why the computeFeps function was designed to return the first replication option(either cartesian or zipped) as soon as the match is found. So I am keeping that the same way

Reply:
For the example in this thread, performance results after the fix(snapshot on top) and before the fix(below snapshot):

![perf-new](https://user-images.githubusercontent.com/43763136/54831523-ddb2e080-4c90-11e9-8861-c8915940022d.PNG)